react/components/SearchResult.js

```diff
@@ -101,6 +110,113 @@ class SearchResult extends Component {
         showLoadingAsOverlay: true,
       })
     }
+
+    document.addEventListener('scroll', this.handleScroll)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('scroll', this.handleScroll)
+  }
+
+  handleScroll = () => {
+    const scroll = this.props.leanMode ? Infinity : window.scrollY
+    const {runtime: { hints: { mobile } }} = this.props
+    
+    if (typeof scroll !== 'number' || !mobile) return
+
+    if (scroll > this.state.scrollValue) this.handleScrollDown()
```

Comment:
The `scroll` can be equal to  `this.state.scrollValue`? if yes, what happens in this situation? If not, you can use ternary conditional